requests_aws4auth/aws4auth.py

```diff
@@ -615,7 +615,16 @@ def get_canonical_headers(cls, req, include=None):
         # in the signed headers, but Requests doesn't include it in a
         # PreparedRequest
         if 'host' not in headers:
-            headers['host'] = urlparse(str(req.url)).netloc.split(':')[0]
+            purl = urlparse(str(req.url))
+            netloc = purl.netloc
+            # Python's http client only includes the port if it is non-default,
+            # see http.client.HTTPConnection.putrequest. The request URL, on the
+            # other hand, might explicitly include it.
+            if (purl.port is not None
```

Comment:
Checking if `purl.netloc` contains `:` may seem more natural here at first sight, but technically, the host could be an IPv6 address. That's why I rather check `port` here. I verified that both Python 2 and 3 return `None` here if no port is given in `req.url`, and an integer otherwise.